decoders/netflow/netflow.go

```diff
@@ -117,6 +118,38 @@ func DecodeTemplateSet(payload *bytes.Buffer) ([]TemplateRecord, error) {
 	return records, nil
 }
 
+func DecodeIPFIXTemplateSet(payload *bytes.Buffer) ([]TemplateRecord, error) {
+	records := make([]TemplateRecord, 0)
+	var err error
+	for payload.Len() >= 4 {
+		templateRecord := TemplateRecord{}
+		err = utils.BinaryDecoder(payload, &templateRecord.TemplateId, &templateRecord.FieldCount)
+		if err != nil {
+			break
+		}
+
+		if int(templateRecord.FieldCount) < 0 {
+			return records, NewErrorDecodingNetFlow("Error decoding TemplateSet: zero count.")
+		}
+
+		fields := make([]Field, int(templateRecord.FieldCount))
+		for i := 0; i < int(templateRecord.FieldCount); i++ {
+			field := Field{}
+			err = utils.BinaryDecoder(payload, &field)
+			if field.Type > math.MaxInt16 {
+				field.Type = field.Type - math.MaxInt16
```

Comment:
This is off by one, you need

    field.Type -= 1 << 15                                                                                  

Because:

    1000000000000001

needs to turn into just '1', so you need to subtract 

    1000000000000000

not

    0111111111111111

Reply:
Thanks, will update. (could've sworn it was in my fork)